### PR TITLE
Update Intel SGX SDK to 2.19.100.3

### PR DESCRIPTION
**Pull request by nick-mobilecoin**
Update to the new SGX SDK that was recently released, https://github.com/intel/linux-sgx/releases/tag/sgx_2.19

### CODE REVIEW

**Comment by nick-mobilecoin** [comment]
Current dependencies on/for this PR:
* master
  * **PR #3252** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/mobilecoin/3252" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/mobilecoinfoundation/mobilecoin/3252?utm_source=stack-comment).

File: consensus/service/BUILD.md [hunk 1]
```diff
@@ -97,8 +97,8 @@ Recommended SDK and package installation:
 (
 	. /etc/os-release
 
-	wget "https://download.01.org/intel-sgx/sgx-linux/2.18/distro/ubuntu${VERSION_ID}-server/sgx_linux_x64_sdk_2.18.100.3.bin"
-	wget "https://download.01.org/intel-sgx/sgx-linux/2.18/distro/ubuntu${VERSION_ID}-server/sgx_linux_x64_driver_2.11.054c9c4c.bin"
+	wget "https://download.01.org/intel-sgx/sgx-linux/2.19/distro/ubuntu${VERSION_ID}-server/sgx_linux_x64_sdk_2.19.100.3.bin"
+	wget "https://download.01.org/intel-sgx/sgx-linux/2.19/distro/ubuntu${VERSION_ID}-server/sgx_linux_x64_driver_2.11.54c9c4c.bin"
```
Comment (nick-mobilecoin) [inline consensus/service/BUILD.md:101]: Note the new URL without the "0"
https://download.01.org/intel-sgx/sgx-linux/2.19/distro/ubuntu22.04-server
versus
https://download.01.org/intel-sgx/sgx-linux/2.18/distro/ubuntu22.04-server

**Comment by nick-mobilecoin** [comment]
I'm a little concerned by the failure at, https://github.com/mobilecoinfoundation/mobilecoin/actions/runs/4441143485/jobs/7795847313
it passed locally and passed on a subsequent run.  I'm not sure if we've seen this failure before or not.

**Review by jcape — APPROVED** [review]

**Review by awygle — APPROVED** [review]
LGTM

**Pull request merged**